projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -229,6 +229,10 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
         });
         esf.columnChange.pipe(takeUntil(this.destroy$)).subscribe(() => {
             this.virtDir?.resetScrollPosition();
+
+            if (this.virtDir) {
+                this.virtDir.state.startIndex = 0;
```

Comment:
I'm not sure why this is necessary. Is perhaps the scrollbar already hidden at the time you are trying to reset it?
I also cannot reproduce the issue as is described in the original issue  steps in 17.1.x:
![filtring](https://github.com/IgniteUI/igniteui-angular/assets/10397980/3b38607e-8674-4657-8cd4-8e48327934fe)

Am I perhaps missing something or are steps slightly different?

Reply:
You can reproduce the issue in the ESF templates sample below. 
Since we're reusing the `IgxGridExcelStyleFilteringComponent` and we do not reset the `startIndex` anymore (from #13688), it doesnt reset when changing the column.

[screen-capture.webm](https://github.com/IgniteUI/igniteui-angular/assets/19392175/f883ea5e-19e7-491b-a074-058783ccbdef)

Notice how the startIndex of the second column is 14.